servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -488,4 +490,16 @@ private static int verifyPortRange(final int port) {
         }
         return port;
     }
+
+    private static void setRequestAuthority(HttpRequestMetaData metaData) {
+        if (metaData.context().containsKey(AUTHORITY_KEY) || metaData.host() == null) {
```

Comment:
We discussed this and #2868 approach offline. My 2c: #2868 feels more natural for multi-address client. Because we already ask users to provide absolute-form URL and internally changing it to relative-form URL for them, it feels natural that the multi-address client should also set a header value because it's the place that has all necessary information available.